packages/strategies/stress.go

```diff
@@ -70,65 +87,220 @@ func (s *StressTest) modifyTransactions(input *types.Block, transactions types.T
 
 	result := types.Transactions{}
 
-	depositTxns := types.Transactions{}
-	userTxns := types.Transactions{}
-
-	for _, txn := range transactions {
-		if txn.IsDepositTx() {
-			depositTxns = append(depositTxns, txn)
-		} else {
-			userTxns = append(userTxns, txn)
-		}
-	}
+	depositTxns, userTxns := s.splitTxs(transactions)
 
 	currentBlockNum := input.NumberU64()
 
 	startBlockNum := s.startBlock.NumberU64()
 	// If first 10 blocks: Add one deposit transactions to send ETH to accounts
 	// Maybe if seqNum=0 (?)
 	if currentBlockNum < startBlockNum+11 {
-		i := currentBlockNum - startBlockNum - 1
-		addr := addresses[i]
+		dep := s.prepareDeposit(l1Info, currentBlockNum)
+		depositTxns = append(depositTxns, dep)
 
-		// Add deposit transaction
-		var dep types.DepositTx
+		if s.cfg.InjectERC20 && currentBlockNum == startBlockNum+1 {
+			erc20, err := s.prepareDeployErc20(input)
+			if err != nil {
+				panic(err)
+			}
 
-		source := derive.UserDepositSource{
-			L1BlockHash: l1Info.BlockHash, // l1 block info
-			LogIndex:    10_000,           // hardcode to a very large number
+			userTxns = append(types.Transactions{erc20}, userTxns...)
 		}
+	} else {
+		if s.cfg.InjectERC20 {
+			if erc20Address == (common.Address{}) {
+				s.waitForFirstBlock()
+				ctx := context.Background()
+				receipt, err := retry.Do(ctx, 3, retry.Exponential(), func() (*types.Receipt, error) {
+					return s.clients.DestNode.TransactionReceipt(ctx, deployTxHash)
+				})
+				if err != nil {
+					panic(err)
+				}
+				erc20Address = receipt.ContractAddress
+			}
+			userTxns = append(userTxns, s.prepareTransferErc20(input)...)
```

Comment:
Could we move this whole `if` block into a new `s.packERC20` method instead? Could refactor `s.packItUp` to make this a little easier to follow:
```
func (s *StressTest) packItUp(input *types.Block) types.Transactions {
  var txs type.Transactions
  if cfg.InjectERC20 {
    txs = s.packERC20(input)
  } else {
    txs = s.packEthTransfer(input)
  }
  return txs
}
```

Reply:
It's a better method, and I've made the changes.